pkg/route/api/validation/validation.go

```diff
@@ -38,7 +40,7 @@ func ValidateRoute(route *routeapi.Route) field.ErrorList {
 		}
 	}
 
-	if err := validateWildcardPolicy(route.Spec.Host, route.Spec.WildcardPolicy, specPath.Child("wildcardPolicy")); err != nil {
+	if err := validateWildcardPolicy(route.Spec.Host, route.Spec.WildcardPolicy, route.Annotations[HostGeneratedAnnotationKey] == "true", specPath.Child("wildcardPolicy")); err != nil {
```

Comment:
its getting set it `pkg/route/registry/route/strategy.go` - validation is called after. We need to still allow that to work for older/non-wildcard routes.